File: backend/internal/domain/transport.go
package domain

import "time"

type TransportType string

const (
	TransportFlight TransportType = "flight"
	TransportTrain  TransportType = "train"
	TransportCar    TransportType = "car"
	TransportBus    TransportType = "bus"
)

type Transport struct {
	ResourceMeta
	TripID         string
	FromLocationID string
	ToLocationID   string
	DepartureTime  *time.Time
	ArrivalTime    *time.Time
	Type           TransportType
	Notes          string
}

File: frontend/types/transport.ts
import { components } from "@/generated/types";

export type TransportResponse = components["schemas"]["TransportResponse"];
export type CreateTransportRequest = components["schemas"]["CreateTransportRequest"];
export type UpdateTransportRequest = components["schemas"]["UpdateTransportRequest"];

export type Transport = {
    fromLocationId: string;
    toLocationId: string;
    departureTime?: string;
    arrivalTime?: string;
    type: "flight" | "train" | "car" | "bus";
    notes?: string;
}